x86_energy_plugin.cpp

```diff
@@ -0,0 +1,627 @@
+            std::string metric_name(name);
+            logging::debug() << "get_event_info called";
+            logging::debug() << "metrics name: " << name;
+            std::vector<scorep::plugin::metric_property> properties;
+
+            /* Allow the user to prefix the metric with x86_energy/ or not.
+             * In the trace it will always be full_name with 
+             * x86_energy/[COREi, PACKAGEi, GPUi] */
+            auto pos_prefix = metric_name.find(prefix_);
+            if (pos_prefix == 0)
+            {
+                metric_name = metric_name.substr(prefix_.length());
+            }
+
+            /* compatible with old plugin interface */
+            auto pos_quantity = metric_name.find("/");
+            auto pos_quantity_old = metric_name.find_last_of("_");
+            std::string quantity;
+            if (metric_name.substr(pos_quantity_old + 1) == "power")
+            {
+                logging::warn() << "use old interface for quantity";
+                quantity="P";
+                metric_name.erase(pos_quantity_old);
+            }
+            else if (metric_name.substr(pos_quantity_old + 1) == "energy")
+            {
+                logging::warn() << "use old interface for quantity";
+                quantity="E";
+                metric_name.erase(pos_quantity_old);
+            }
+            else if (pos_quantity == std::string::npos)
+            {
+                logging::warn() << "no physical quantity found using Energy";
+                quantity="E";
+            }
+            else
+            {
+                quantity=metric_name.substr(pos_quantity+1);
+                metric_name.erase(pos_quantity);
+            }
+
+            /* also match metric name core for core0 and core1 */
+            scorep::plugin::util::matcher match(metric_name + "*");
+            for (int i = 0; i < x86_energy.nr_packages(); i ++)
+            {
+                for (int j = 0; j < x86_energy.features(); j++)
+                {
+                    std::string sensor_name(x86_energy.name(j)
+                            + std::to_string(i));
+                    /* converting sensor name to upper case */
+                    std::transform(sensor_name.begin(), sensor_name.end(), 
+                            sensor_name.begin(), ::toupper);
+                    logging::debug() << "found sensor: " << sensor_name;
+                    logging::debug() << match(sensor_name);;
+
+                    if (match(sensor_name))
+                    {
+                        properties.push_back(add_metric_property(sensor_name, 
+                                    x86_energy.ident(j), i, quantity));
+                    }
+                }
+            }
+
+            if (properties.empty())
+            {
+                logging::fatal()<< "No metrics added. Check your metrics!";
+            }
+            /* add blade counter if necessary */
+            else if (match("BLADE") or match("blade"))
```

Comment:
BLADE should not be included in "*"! It is highly specific and only useful if an actual determined offset is given.